api/rest/execKubeConfig.go

```diff
@@ -48,7 +52,13 @@ func HandleKubeConfig(c *gin.Context) {
 		return
 	}
 
-	err := HandleKubeConfigCreation(c, &initConfigParams, token)
+	execRequest := handleContainerResolve(c, token, initConfigParams.ContainerName)
+	if execRequest == nil {
+		rest.WriteResponse(c, http.StatusInternalServerError, "Could not retrieve exec request")
+		return
+	}
+
+	err := HandleKubeConfigCreation(&kubeConfigParams, execRequest)
```

Comment:
As far as I understand you dropped token argument to make args list shorter, but now you kind of re-purposing kubeConfigParams.BearerToken, because in addition to propagate into kubeconfig, you also use it to create K8s API and make API call.
Ideally, I think K8sAPI should be created per request(on demand since there is requests in exec manager which does not need it) and reused in places where it's needed, without need to propagate the token.

If I made this PR I would introduce one more argument without dropping token, but I'm not against the made change here as well.

Reply:
Perhaps I misunderstood the existing code -- my understanding is that token is derived from the same value, i.e.
- if `initConfigParams.KubeConfigParams.BearerToken` is set, it is used
- if it's unset, we log in via username/password and get the token

Both current flows explictly set `initConfigParams.BearerToken = token` in `HandleKubeConfigCreation`, so I thought I woudl move this step earlier in the call stack and avoid having to pass initConfigParams and the token only to apply the token to initConfigParams.